app/templates/agenda.html

```diff
@@ -3,32 +3,47 @@
 {% block head %}
   {{ super() }}
   <link href="{{ url_for('static', path='/agenda_style.css') }}" rel="stylesheet">
+  <script type="text/javascript" src="{{ url_for('static', path='js/categories_filter.js') }}"></script>
 {% endblock %}
 
 {% block content %}
-  <form method="GET" action="/agenda#dates" class="mx-3 pt-3">
-    <div class="col-sm-3">
-      <label for="start_date">{{ gettext("From") }}</label>
-      <input class="form-control" type="date" id="start_date" name="start_date" value='{{ start_date }}'>
+  <div class="agenda_grid">
+    <div class="agenda_filter_grid">
+      <form method="GET" action="/agenda#dates" class="mx-3 pt-3">
+        <div class="col-sm-3">
+          <label for="start_date">{{ gettext("From") }}</label>
+          <input class="form-control" type="date" id="start_date" name="start_date" value='{{ start_date }}'>
+        </div>
+        <div class="col-sm-3">
+          <label for="end_date">{{ gettext("To") }}</label>
+          <input class="form-control" type="date" id="end_date" name="end_date" value='{{ end_date }}'><br>
+        </div>
+        <div>
+          <input class="btn btn-primary btn-sm" type="submit" value="{{ gettext('Get Agenda') }}">
+        </div>
+      </form>
     </div>
-    <div class="col-sm-3">
-      <label for="end_date">{{ gettext("To") }}</label>
-      <input class="form-control" type="date" id="end_date" name="end_date" value='{{ end_date }}'><br>
-    </div>
-    <div>
-      <input class="btn btn-primary btn-sm" type="submit" value="{{ gettext('Get Agenda') }}">
-    </div>
-  </form>
 
-  <div class="exact_date pt-3">
-    <div class="mx-3">
-      <a class="btn btn-light btn-outline-primary btn-sm" href="/agenda?days=0#dates">{{ gettext("Today") }}</a>
+    <div class="category_filter">
+      <label for="category" class="sr-only">Filter events by category</label>
+      <input type="text" id="category" name="category" class="form-control-sm"
+             placeholder="category ID" value="{{ category }}"
+             onfocus="this.value=''" required>
+      <button class="btn btn-outline-info btn-sm" onclick='filterByCategory({{ events | tojson | safe }})'>
```

Comment:
Prefer using JS's `addEventListener` to do this (we do try to separate the view from the logic)